addons/mail_bot/models/mail_bot.py

```diff
@@ -11,26 +11,28 @@
 class MailBot(models.AbstractModel):
     _description = 'Mail Bot'
 
-    def _apply_logic(self, record, values, command=None):
+    def _apply_logic(self, channel, values, command=None):
         """ Apply bot logic to generate an answer (or not) for the user
         The logic will only be applied if odoobot is in a chat with a user or
         if someone pinged odoobot.
 
-         :param record: the mail_thread (or discuss_channel) where the user
-            message was posted/odoobot will answer.
+         :param channel: the discuss channel where the user message was posted/odoobot will answer.
          :param values: msg_values of the message_post or other values needed by logic
          :param command: the name of the called command if the logic is not triggered by a message_post
         """
+        channel.ensure_one()
         odoobot_id = self.env['ir.model.data']._xmlid_to_res_id("base.partner_root")
-        if len(record) != 1 or values.get("author_id") == odoobot_id or values.get("message_type") != "comment" and not command:
+        if values.get("author_id") == odoobot_id or values.get("message_type") != "comment" and not command:
             return
-        if self._is_bot_pinged(values) or self._is_bot_in_private_channel(record):
```

Comment:
Note: cleanup, condition `_is_bot_pinged` irrelevant as `_is_bot_in_private_channel` was always checked in `_get_answer`.